tests/testthat/test-adjust-time.R

```diff
@@ -1,48 +1,61 @@
-test_that("adjust time test", {
-  extracted_feature_files <- c('../testdata/adjust-time/extracted_0.parquet',
-                               '../testdata/adjust-time/extracted_1.parquet',
-                               '../testdata/adjust-time/extracted_2.parquet')
+patrick::with_parameters_test_that(
+  "adjust time test",
+  {
+    skip_on_ci()
+    
+    testdata <- file.path("..", "testdata")
+    
+    filenames <- lapply(files, function(x) {
+      file.path(testdata, x, paste0(x, "_features.Rds"))
+    })
+    
+    extracted <- lapply(filenames, readRDS)
+    
+    chk <- Sys.getenv("_R_CHECK_LIMIT_CORES_", "")
+    
+    if (nzchar(chk) && chk == "TRUE") {
+      # use 2 cores in CRAN/Travis/AppVeyor
+      cluster <- 2L
+    } else {
+      # use all cores in devtools::test()
+      cluster <- parallel::detectCores()
+    }
+    
+    if (!is(cluster, "cluster")) {
+      cluster <- parallel::makeCluster(cluster)
+      on.exit(parallel::stopCluster(cluster))
+    }
+    
+    # NOTE: side effect (doParallel has no functionality to clean up)
+    doParallel::registerDoParallel(cluster)
+    
+    corrected <- adjust.time(
+      features = extracted,
+      mz_tol_relative = mz_tol,
+      rt_tol_relative = chr_tol,
+      mz_max_diff = find_tol_max_d,
+      mz_tol_absolute = max_align_mz_diff,
+      do.plot = FALSE
+    )
+    
+    expected_filenames <- lapply(files, function(x) {
+      file.path(testdata, "adjust-time", paste0(x, "_features_corrected.parquet"))
+    })
+    
+    expected <- lapply(expected_filenames, arrow::read_parquet)
+    expected <- lapply(expected, as.data.frame)
+    
+    corrected <- lapply(corrected, as.data.frame)
   
-  corrected_files <- c('../testdata/adjust-time/corrected_0.parquet',
-                       '../testdata/adjust-time/corrected_1.parquet',
-                       '../testdata/adjust-time/corrected_2.parquet')
-
-  chk <- Sys.getenv("_R_CHECK_LIMIT_CORES_", "")
-  
-  if (nzchar(chk) && chk == "TRUE") {
-    # use 2 cores in CRAN/Travis/AppVeyor
-    cluster <- 2L
-  } else {
-    # use all cores in devtools::test()
-    cluster = 4
-  }
-
-  if (!is(cluster, 'cluster')) {
-    cluster <- parallel::makeCluster(cluster)
-    on.exit(parallel::stopCluster(cluster))
-  }
-
-  # NOTE: side effect (doParallel has no functionality to clean up)
-  doParallel::registerDoParallel(cluster)
-
-  extracted <- lapply(extracted_feature_files, arrow::read_parquet)
-  extracted <- lapply(extracted, as.matrix)
-
-  expected <- lapply(corrected_files, arrow::read_parquet)
-  expected <- lapply(expected, as.matrix)
-
-  corrected <- adjust.time(
-    features = extracted,
-    mz.tol = NA,
-    chr.tol = NA,
-    find.tol.max.d = 10 * 1e-05,
-    max.align.mz.diff = 0.01,
-    do.plot = FALSE
+    expect_equal(corrected, expected)
```

Comment:
So here we are comparing the files as they are, with no tolerances etc. Compared to tests of `extract features`, this is less robust. @hechth do you think this is enough in this case?

Reply:
Yeah - there seems to be no problem related to probrabilities etc. in this part, it seems to be only in the extract features part so far.